### ISSUE TITLE
warning: "__ARM_FP" is not defined

### ISSUE THREAD

**Issue by me-axentia** [ARM-software/CMSIS_6#120]
With the new CMSIS 6.0.0 I get this warning when compiling with GCC with soft float:
CMSIS/include/m-profile/cmsis_gcc_m.h:1535:6: warning: "__ARM_FP" is not defined, evaluates to 0 [-Wundef]

As far as I know __ARM_FP is only defined if hard float is used.
core_cm4.h checks if it is defined `#if defined (__ARM_FP)` but in two places in cmsis_gcc_m.h `__ARM_FP` is used without checking if it is defined. For example:
```
__STATIC_FORCEINLINE void __set_FPSCR(uint32_t fpscr)
{
#if (__ARM_FP >= 1)
  __builtin_arm_set_fpscr(fpscr);
#else
  (void)fpscr;
#endif
}
```
I consider this a bug since this generates a lot of warnings (if not suppressed). (This applies to all compilers and architectures.)

**Comment by JonatanAntoni** [comment]
Hi @me-axentia,

Thanks for pointing this out.
This seems to be an oversight.
This should rather be `#if (defined(__ARM_FP) && (__ARM_FP >= 1))`.
I guess this should fix the bug. Can you confirm?

**Comment by me-axentia** [comment]
> Hi @me-axentia,
> 
> Thanks for pointing this out. This seems to be an oversight. This should rather be `#if (defined(__ARM_FP) && (__ARM_FP >= 1))`. I guess this should fix the bug. Can you confirm?

Yes, this fixes the issue.

**Comment by JonatanAntoni** [comment]
Hi @me-axentia,

we've just committed the fix. It will be part of the next release.
Please consider closing this issue if its solved for you.

Thanks,
Jonatan

**Comment by me-axentia** [comment]
> Hi @me-axentia,
> 
> we've just committed the fix. It will be part of the next release. Please consider closing this issue if its solved for you.
> 
> Thanks, Jonatan

Hello. It is solved for me but not for the a-profile. For example this file is not fixed: Include/a-profile
/cmsis_gcc_a.h
It is also not fixed for the other compilers (I don't know if they omit warning for this) in the other files in the same folder.

**Comment by JonatanAntoni** [comment]
Would you be happy to contribute a PR fixing all the locations you found?

**Comment by GuentherMartin** [comment]
@JonatanAntoni , please see https://github.com/ARM-software/CMSIS_6/pull/124

**Comment by JonatanAntoni** [comment]
Fixed by #124
